tests/js/server/dump/dump-test.inc

```diff
@@ -835,6 +835,51 @@
         assertEqual(1, res.length);
       },
 
+      ////////////////////////////////////////////////////////////////////////////////
+      /// @brief test search-alias restoring
+      ////////////////////////////////////////////////////////////////////////////////
+
+      testSearch: function () {
+        let collection = db._collection("searchCollection");
+        assertNotEqual(null, collection);
+
+        let index = collection.index("searchIndex");
+        assertTrue(index.includeAllFields);
+        assertEqual(index.consolidationIntervalMsec, 0);
+        assertEqual(index.cleanupIntervalStep, 42);
+
+        let search = db._view("searchSearch");
+        assertNotEqual(null, search);
+        assertEqual("searchSearch", search.name());
+        let props = search.properties();
+        assertEqual(props.indexes, [{ collection: "searchCollection", index: "searchIndex" }]);
+
+        // After a restore, the Inverted index needs to be rebuilt, therefore
+        // we cannot expect that the data is immediately visible. We can only
+        // hope that the data will be there in due course:
+
+        var startTime = new Date();
+        var res;
+        while (new Date() - startTime < 60000) {
+          res = db._query("FOR doc IN " + search.name() + " SEARCH doc.value >= 0 OPTIONS { waitForSync: true } RETURN doc").toArray();
+          if (res.length === 5000) {
+            break;
+          }
+          console.log("Waiting for inverted index to be built...");
+          internal.wait(1);
+        }
```

Comment:
we don't need this anymore in presence of waitForSync
```
db._query("FOR doc IN " + search.name() + " SEARCH doc.value >= 0 OPTIONS { waitForSync: true } RETURN doc")
```

Reply:
Old code. I will change it if build failed